src/frontend/mosh-server.cc

```diff
@@ -549,14 +582,31 @@ static void serve( int host_fd, Terminal::Complete &terminal, ServerConnection &
 
   while ( 1 ) {
     try {
+      static const uint64_t timeout_if_no_client = 60000;
+      int timeout = INT_MAX;
       uint64_t now = Network::timestamp();
 
-      const int timeout_if_no_client = 60000;
-      int timeout = min( network.wait_time(), terminal.wait_time( now ) );
+      timeout = min( timeout, network.wait_time() );
+      timeout = min( timeout, terminal.wait_time( now ) );
```

Comment:
Note the min() on both lines.  The intent is to find the lowest timeout value from several possible wait periods.  I believe it's correct.